pkg/controllerhelpers/handlers.go

```diff
@@ -93,6 +93,16 @@ func (h *Handlers[T]) EnqueueAll(depth int, untypedObj kubeinterfaces.ObjectInte
 	h.EnqueueAllFunc(h.Enqueue)(depth+1, untypedObj, op)
 }
 
+func (h *Handlers[T]) EnqueueAllWithUntypedFilterFunc(filterFunc func(kubeinterfaces.ObjectInterface) bool) EnqueueFuncType {
```

Comment:
you'd cover more use cases with

```
func (h *Handlers[T]) EnqueueWithUntypedFilterFunc(filterFunc func(kubeinterfaces.ObjectInterface) bool) EnqueueFuncType {
	return func(depth int, untypedObj kubeinterfaces.ObjectInterface, op HandlerOperationType) {
		if !filterFunc(untypedObj) {
			return
		}

		h.Enqueue(depth+1, untypedObj, op)
	}
}
```

which you can compose to have the same functionality

```
ncc.handlers.EnqueueAllFunc(ncc.handlers.EnqueueWithUntypedFilterFunc(isManagedByNodeConfigController))
```

Reply:
notice the different argument types, those do not compose

Reply:
copy-pasted from my comment https://github.com/rzetelskik/scylla-operator/commit/8c8f4b4f4fa616d3dbdeed26e869e7b980a8c5f2, compiles for me

Reply:
ok, my bad, these are not equivalent